tests/components/tplink_omada/test_binary_sensor.py

```diff
@@ -0,0 +1,143 @@
+"""Tests for TP-Link Omada sensor entities."""
+
+from datetime import timedelta
+import json
+from unittest.mock import MagicMock, patch
+
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+from syrupy.assertion import SnapshotAssertion
+from tplink_omada_client.definitions import DeviceStatus, DeviceStatusCategory
+from tplink_omada_client.devices import OmadaListDevice
+
+from homeassistant.components.tplink_omada.const import DOMAIN
+from homeassistant.components.tplink_omada.coordinator import POLL_DEVICES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import (
+    MockConfigEntry,
+    async_fire_time_changed,
+    load_fixture,
+    snapshot_platform,
+)
+
+POLL_INTERVAL = timedelta(seconds=POLL_DEVICES)
+
+
+@pytest.fixture
+async def init_integration(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_omada_client: MagicMock,
+) -> MockConfigEntry:
+    """Set up the TP-Link Omada integration for testing."""
+    mock_config_entry.add_to_hass(hass)
+
+    with patch("homeassistant.components.tplink_omada.PLATFORMS", ["binary_sensor"]):
+        assert await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    return mock_config_entry
```

Comment:
You already have this in conftest.py. You could just add a small autoused fixture that just patches the platforms here

Reply:
I learned something today about autoused fixtures and generator fixtures.